### PR TITLE
Add support for target designs with SerialBridge but no DRAM

### PR DESCRIPTION
**Pull request by jerryz123**
Support target designs with no AXI4 backing DRAM. The changes modify firesim_fesvr to not use the loadmem channel if it is unavailable.

There will be a corresponding Chipyard PR when millennium is back.

### CODE REVIEW

File: sim/midas/src/main/scala/midas/widgets/UsesHostDRAM.scala [hunk 1]
```diff
@@ -49,7 +49,9 @@ trait HostDramHeaderConsts {
    * to give it an independent region.
    *
    */
-  def memoryRegionName: String
+  def memoryRegionNameOpt: Option[String]
```
Comment (davidbiancolin) [inline sim/midas/src/main/scala/midas/widgets/UsesHostDRAM.scala:52]: Can you explain the rationale for this change. It has knock on effects for DRAM allocation.

Comment (jerryz123) [reply sim/midas/src/main/scala/midas/widgets/UsesHostDRAM.scala:52]: The `HostDramHeaderConsts` trait is used by `SerialBridgeModule`, which does not have a `memoryRegionName` if there is no target DRAM.

Comment (davidbiancolin) [reply sim/midas/src/main/scala/midas/widgets/UsesHostDRAM.scala:52]: I think in that case what we should do is remove the mixin from the serial bridge; and directly add an option member there. 

If we want bridges to be able to inspect the dram allocation map, i think we can push something into the parameters instance before FPGATop module instantiation.

Comment (davidbiancolin) [reply sim/midas/src/main/scala/midas/widgets/UsesHostDRAM.scala:52]: Ill wait for the rest of the CY PR though.

File: sim/firesim-lib/src/main/scala/bridges/SerialBridge.scala [hunk 2]
```diff
@@ -10,16 +10,18 @@ import freechips.rocketchip.config.Parameters
 
 import testchipip.{SerialIO, SerialAdapter}
 
-class SerialBridge(memoryRegionName: String) extends BlackBox with Bridge[HostPortIO[SerialBridgeTargetIO], SerialBridgeModule] {
+case class SerialBridgeParams(memoryRegionNameOpt: Option[String])
```
Comment (davidbiancolin) [inline sim/firesim-lib/src/main/scala/bridges/SerialBridge.scala:13]: Can you add scaladoc for this explaining how it's used.

**Review by davidbiancolin — APPROVED** [review]
LGTM. Remember to add the dev->master PR message.

File: sim/midas/src/main/scala/midas/widgets/UsesHostDRAM.scala (hunk 1 above)
Comment (jerryz123) [reply sim/midas/src/main/scala/midas/widgets/UsesHostDRAM.scala]: CY PR is here: https://github.com/ucb-bar/chipyard/pull/682

**Pull request merged**